### ISSUE TITLE
No decrease in loss when using model without chain

### ISSUE THREAD

**Issue by ashishkumar1550** [FluxML/model-zoo#29]
I am using the same code as in mlp.jl, I kept the same model without using chain, looks like this:

l1 = Dense(28^2,32, relu)
l2 = Dense(32, 10)
m(x) = softmax(l2(l1(x)))

now whenever I run the model the loss doesn't decrease on training.

**Comment by iblis17** [comment]
Since `m(x)` is a function, but `Chain` is a struct.
`params(mychain)` will tell optimizer which parameters you want to train.
And, in you case, `params(function)` won't work.

**Comment by ashishkumar1550** [comment]
Thanks
